### ISSUE TITLE
Embedded server creates auth file even if authentication is disabled.

### ISSUE THREAD

**Issue by DirkMahler** [neo4j/neo4j#4209]
jQAssistant starts an embedded Neo4j server using WrappingNeoServer. Currently I'm preparing migration to Neo4j 2.2 (RC1) and I need to disable authentication:

```
configuration.setProperty(ServerSettings.auth_enabled.name(), Boolean.FALSE.toString());
```

The running instance behaves like expected but in the working directory of the Java process a file "data/dbms/auth" is created containing the default credentials of Neo4j:

```
neo4j:SHA-256,D32B7BA9BBBE154901C041AEBDC62E84FD79395E7C9F5D8696515D84F7FB1641,BF7E9F3FA4CF296758E46C20A4623ED8:password_change_required
```

**Comment by cleishm** [comment]
Hey @DirkMahler,

I'm not sure what you mean by an _embedded_ Neo4j server. Neo4j Server is a standalone application, and thus not embedded. The class you've mentioned, `WrappingNeoServer`, is an internal component of that application and not really intended for direct use by external applications.

Is there a use-case you're trying to achieve? Can you not run Neo4j Server independently?

Cheers,
Chris

**Comment by DirkMahler** [comment]
Hi @cleishm,

jQAssistant (http://jqassistant.org) is a static code analysis tool like Checkstyle or FindBugs which can be integrated into a build prozess, e.g. as a Maven plugin. A developer shall not be forced to care about installation and configuration of a independent server instance or even upgrading it - things must work out of the box to avoid acceptance problems. The only thing to get started is this:

```
mvn jqassistant:scan
```
Maven will do the rest (e.g. downloading all required artifacts, including Neo4j libs).

Another important aspect is performance: the time impact of a code analysis tool shall be as low as possible, during a quite short time thousands or even millions of nodes/relationships must be created and/or traversed - any remote protocol would become a problem here. It's again about acceptance by developers.

Therefore I decided to use the Java API as described in the Neo4j documentation (http://neo4j.com/docs/milestone/tutorials-java-embedded.html). Currently jQAssistant does not provide its own UI - there are ideas for that but for a start the Neo4j browser is just a real good thing. So I decided to also use it within jQAssistant, so for exploring your code structures (e.g. Java classes, packages, Maven modules, Java EE descriptors etc.) a developer just enters

```
mvn jqassistant:server
```
and opens his web browser on the url http://localhost:7474

I hope these explanations help to explain my needs. If you want to get an idea about the structures which are scanned by jQAssistant have a look at the running demo instance (http://jqassistant.org/get-started/ Explore Java Structures online). On my notebook it currently takes about 2 minutes.

**Comment by cleishm** [comment]
Hey @DirkMahler,

Thanks for going into that detail. I've seen JQAssistant, and think what you've done there is awesome. And it's a great use of Neo4j.

I'm going to see if we can make Neo4j Server not create that file when auth is disabled, as a one off and on the assumption it doesn't turn out to be too complicated. However, I also suggest that you think about how to make JQAssistant work with a downloaded and installed Neo4j Server, as it's highly likely that these problems will keep occurring and it'll become more and more difficult to directly use the server java classes. As I mentioned, they exist for the purpose of building a standalone Neo4j Server application, and as the server evolves, they'll continue to be developed and modified according to those needs.

Cheers,
Chris

**Comment by DirkMahler** [comment]
Thanks Chris!

I think it will be diffcult for me to go away from using the GraphDatabaseService directly - but I see the potential and some need to (sooner or later) move away to a jQAssistant specific UI implementation.

BTW: Without Neo4j providing all this flexibility and performance the implementation of software analytics stuff won't be possible "that easy" - Great work from your side!

**Comment by cleishm** [comment]
Resolved by #4245
